addons/l10n_eu_oss/models/res_company.py

```diff
@@ -111,28 +111,43 @@ def _get_repartition_lines_oss(self):
     def _get_oss_account(self):
         self.ensure_one()
         if not self.env.ref(f'l10n_eu_oss.oss_tax_account_company_{self.id}', raise_if_not_found=False):
+            self._create_oss_account()
+        return self.env.ref(f'l10n_eu_oss.oss_tax_account_company_{self.id}')
+
+    def _create_oss_account(self):
+        known_oss_accounts = {
+            'de_skr03': 'account_1767',
+            'de_skr04': 'chart_skr04_3817',
+        }
+        if (self.chart_template in known_oss_accounts
+            and (oss_account_if_exists :=
+                self.env['account.chart.template'].with_company(self).ref(known_oss_accounts[self.chart_template],
+                                                                          raise_if_not_found=False))):
+            oss_account = oss_account_if_exists
+        else:
             sales_tax_accounts = self.env['account.tax'].search([
                     *self.env['account.tax']._check_company_domain(self),
                     ('type_tax_use', '=', 'sale'),
                 ]).invoice_repartition_line_ids.mapped('account_id')
             if not sales_tax_accounts:
                 return False
-            new_code = self.env['account.account']._search_new_account_code(self, len(sales_tax_accounts[0].code), sales_tax_accounts[0].code[:-2])
+            new_code = self.env['account.account']._search_new_account_code(sales_tax_accounts[0].code,
+                                                                            self,
+                                                                            cache=[sales_tax_accounts[0].code])
             oss_account = self.env['account.account'].create({
                 'name': f'{sales_tax_accounts[0].name} OSS',
                 'code': new_code,
                 'account_type': sales_tax_accounts[0].account_type,
                 'company_id': self.id,
                 'tag_ids': [(4, tag.id, 0) for tag in sales_tax_accounts[0].tag_ids],
                 })
-            self.env['ir.model.data'].create({
-                'name': f'oss_tax_account_company_{self.id}',
-                'module': 'l10n_eu_oss',
-                'model': 'account.account',
-                'res_id': oss_account.id,
-                'noupdate': True,
-                })
-        return self.env.ref(f'l10n_eu_oss.oss_tax_account_company_{self.id}')
+        self.env['ir.model.data'].create({
+            'name': f'oss_tax_account_company_{self.id}',
+            'module': 'l10n_eu_oss',
+            'model': 'account.account',
+            'res_id': oss_account.id,
+            'noupdate': True,
+            })
```

Comment:
Since you are already changing this line, let's fix it
```suggestion
        })
```